crates/sui-json-rpc/src/api.rs

```diff
@@ -255,6 +255,34 @@ pub trait RpcTransactionBuilder {
         gas_budget: u64,
     ) -> RpcResult<TransactionBytes>;
 
+    #[method(name = "pay_sui")]
```

Comment:
Sorry commented in the wrong place before. Can we also add some comments for what this endpoint does on line 241, 258 and 273?